packages/api/src/middlewares/RouterMiddlewareFactory.ts

```diff
@@ -56,27 +54,16 @@ export class RouterMiddlewareFactory implements IMiddlewareFactory {
             ctx.status = 200;
             ctx.state.result = "";
           } else {
-            // method not supported
-            ctx.status = 405;
-            ctx.state.result = `${method} not allowed.`;
+            throw new MethodNotSupportedError();
           }
         } else if (error === "ROUTE_NOT_SUPPORTED") {
```

Comment:
This stuff could probably be refactored. What if lookupRoute just threw now that we have all these nice errors now?